applications/asset_tracker_v2/src/events/app_module_event.c

```diff
@@ -61,23 +61,32 @@ static char *get_evt_type_str(enum app_module_event_type type)
 	}
 }
 
+#define LOG_EVENT_BUF_SIZE 50
 static void log_event(const struct app_event_header *aeh)
 {
 	const struct app_module_event *event = cast_app_module_event(aeh);
-	char data_types[50] = "\0";
+	uint16_t size = 0;
+	char data_types[LOG_EVENT_BUF_SIZE] = "\0";
 
 	if (event->type == APP_EVT_ERROR) {
 		APP_EVENT_MANAGER_LOG(aeh, "%s - Error code %d",
 				get_evt_type_str(event->type), event->data.err);
 	} else if (event->type == APP_EVT_DATA_GET) {
 		for (int i = 0; i < event->count; i++) {
-			strcat(data_types, type2str(event->data_list[i]));
+			strncat(data_types, type2str(event->data_list[i]),
+					LOG_EVENT_BUF_SIZE - size - 1);
+			size += strlen(type2str(event->data_list[i]));
 
-			if (i == event->count - 1) {
+			if (i == event->count - 1 || size > LOG_EVENT_BUF_SIZE - 2) {
 				break;
 			}
 
-			strcat(data_types, ", ");
+			strncat(data_types, ", ", LOG_EVENT_BUF_SIZE - size - 1);
+			size += 2;
```

Comment:
```suggestion
			size += strlen(", ");
```

Reply:
I'd use `sizeof` instead of `strlen` calculate this compile time

Reply:
Using `strlen` since `sizeof` results in 3 due to the null character. Also changed `", "` to a define in case someone wants to more easily change the separator in the future.

Reply:
Yeah, `sizeof` includes the null termination, so it's common to subtract 1 to achieve the same as `strlen`. So this would be
```
size += (sizeof(", ") - 1);
```
It's not a big deal, though, but shaves off a few CPU cycles to do this computation compile time.

Reply:
Good point on the optimization. Fixed it.